# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.h
# Commit: 20a5c1f3b35e
# Full Hash: 20a5c1f3b35e80849e0b98ffbe3f9488d703717b
# Author: Frederic Wang <fred.wang@free.fr>
# Date: 2023-11-22 21:50:37
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1807253 - unreliable timing for content-visibility:auto, r=emilio
#   
#   Rewrite implementation of content-visibility: auto as defined in
#   https://github.com/w3c/csswg-drafts/issues/8542
#   
# ==============================================================================

diff -r b90083f581e1 -r 20a5c1f3b35e layout/base/PresShell.h
--- a/layout/base/PresShell.h	Mon Nov 20 11:33:16 2023 +0000
+++ b/layout/base/PresShell.h	Wed Nov 22 13:00:36 2023 +0000
@@ -1748,11 +1748,25 @@
   void RegisterContentVisibilityAutoFrame(nsIFrame* aFrame) {
     mContentVisibilityAutoFrames.Insert(aFrame);
   }
+  bool HasContentVisibilityAutoFrames() const {
+    return !mContentVisibilityAutoFrames.IsEmpty();
+  }
 
   void UpdateRelevancyOfContentVisibilityAutoFrames();
   void ScheduleContentRelevancyUpdate(ContentRelevancyReason aReason);
   void UpdateContentRelevancyImmediately(ContentRelevancyReason aReason);
 
+  // Determination of proximity to the viewport.
+  // Refer to "update the rendering: step 14", see
+  // https://html.spec.whatwg.org/#update-the-rendering
+  struct ProximityToViewportResult {
+    bool mHadInitialDetermination = false;
+    bool mAnyScrollIntoViewFlag = false;
+  };
+  ProximityToViewportResult DetermineProximityToViewport();
+
+  void ClearTemporarilyVisibleForScrolledIntoViewDescendantFlags() const;
+
  private:
   ~PresShell();
 